src/main/java/pl/cleankod/exchange/provider/nbp/errors/NbpErrorDecoder.java

```diff
@@ -0,0 +1,16 @@
+package pl.cleankod.exchange.provider.nbp.errors;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+
+public class NbpErrorDecoder implements ErrorDecoder {
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
```

Comment:
The response details are sunk at this point. This would result in a very generic log entry: `500 - deal with it`. 😉